### ISSUE TITLE
Reply All replies to self in email chain

### ISSUE THREAD

**Issue by patrickheeney** [Foundry376/Mailspring#206]
##### Are there any related issues?
...

##### What operating system are you using?
Mac OSX

##### What version of Mailspring are you using?
1.0.5

--

**Bug?**
##### Do you have any third-party plugins installed? If so, which ones?
...

##### Is the issue related to a specific email provider (Gmail, Exchange, etc.)?
Gmail?

##### Is the issue reproducible with a particular attachment, message, signature, etc?
...

I sent an initial reply to an email that came in. I wanted to add more information so I went to reply to my initial reply. The button shows as Reply all even though there are two parties (myself sending to them). 

![screen shot 2017-10-19 at 10 00 17 am](https://user-images.githubusercontent.com/1407228/31783792-5a2eea1c-b4b4-11e7-99e0-171a24c1858f.png)

Clicking reply all causes a new email thread to get sent from myself to myself with them as a CC.

![screen shot 2017-10-19 at 10 00 24 am](https://user-images.githubusercontent.com/1407228/31783843-8911f0ea-b4b4-11e7-8156-6102180aa108.png)

I expected the replies to be sent from myself to them directly without reply all or without sending to myself and ccing them.

**Comment by palefire19** [comment]
+1 for this one

**Comment by bengotow** [comment]
I think I actually saw this the other day - let me see if I can get this fixed.

**Comment by palefire19** [comment]
Ben, note it's not just reply-all, it's also a regular reply. If I was the last person to send an email in an email chain, reply emails just me and reply-all includes me. Both are not the expected behavior.

**Comment by bengotow** [comment]
Hey folks - thanks for the additional detail. It turns out this happened whenever the last message in the thread was sent by you. Mailspring now adds the replyTo header all the time (I think it's a mailcore2 thing), and Mailspring was seeing that replyTo header, using it, and addressing any follow-up replies or reply-alls to you instead of the correct recipients.

I've committed a change so that Mailspring ignores the replyTo field in this case. I've also added some more tests so this won't happen again.
